openmdao/core/tests/test_group.py

```diff
@@ -489,6 +489,46 @@ def test_reused_output_promoted_names(self):
         with self.assertRaisesRegex(Exception, msg):
             prob.setup()
 
+    def test_unconnected_input_required_connection(self):
```

Comment:
I didn't see it but might be good to have a test for when the variable isn't connected but is declared as a design variable.